### PR TITLE
Add NotExport-plugin not to export inline-c-ffi's functions

### PR DESCRIPTION
**Pull request by junjihashimoto**
Functions generated by inline-c may conflict.
This PR provides a plugin to remove functions having the prefix string of `inline_c_ffi` from the list of exports.

We can see the result from this log.
https://github.com/hasktorch/hasktorch/pull/612/checks?check_run_id=3318413468

### CODE REVIEW

File: hasktorch/hasktorch.cabal [hunk 1]
```diff
@@ -96,10 +96,11 @@ library
  other-modules:       Paths_hasktorch
  hs-source-dirs:      src
  default-language:    Haskell2010
- ghc-options:         -fplugin GHC.TypeLits.Normalise -fplugin GHC.TypeLits.KnownNat.Solver -fplugin GHC.TypeLits.Extra.Solver -fconstraint-solver-iterations=0
+ ghc-options:         -fplugin GHC.TypeLits.Normalise -fplugin GHC.TypeLits.KnownNat.Solver -fplugin GHC.TypeLits.Extra.Solver -fconstraint-solver-iterations=0 -fplugin GHC.NotExport.Plugin
```
Comment (stites) [inline hasktorch/hasktorch.cabal:99]: this seems like a typo -- shouldn't this only be on libtorch-ffi?

Comment (junjihashimoto) [reply hasktorch/hasktorch.cabal:99]: It is not a typo.
Torch.Vision uses inline-c, so I add it.

Comment (junjihashimoto) [reply hasktorch/hasktorch.cabal:99]: Another module will use inline-c.
It is a preventive measure.
https://github.com/hasktorch/hasktorch/pull/600

**Review by tscholak — APPROVED** [review]
LGTM

**Comment by junjihashimoto** [comment]
Thx!

**Comment by junjihashimoto** [comment]
Recompile seems to happen frequently.
image

**Comment by junjihashimoto** [comment]
https://mpickering.github.io/posts/2018-08-10-plugins-recompilation.html

**Comment by junjihashimoto** [comment]
https://hackage.haskell.org/package/ghc-lib-parser-8.8.2/docs/src/Plugins.html#defaultPlugin
impurePlugin is a default.

**Comment by junjihashimoto** [comment]
The plugin becomes a pure plugin.
image

**Pull request merged**